### PR TITLE
Activity 4.1.2] hashtables fixed

### PR DESCRIPTION
**Pull request by JeffTheAggie**
Labels: Difficulty = Medium, Issue = Context and Ordering

Closes Issues #517 .

Changes proposed in this pull request:

- Split Card 6 into 2 cards.

- Differentiated Hashtables and Dictionaries from each other.

@sarahg500

### CODE REVIEW

File: Module4.1_Intro_to_Data_Structures_and_Algos/activities/Act2_HashTables/8.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+However, when we talk about **hash tables**, we're actually talking about a type of **dictionary**. In Python, the Dictionary data types in some way, represent the implementation of hash tables. The Keys in the dictionary satisfy the following requirements.
+
+- The keys of the dictionary are hashable i.e. the are generated by hashing function which generates unique result for each unique value supplied to the hash function.
+- The order of data elements in a dictionary is not fixed.
+
+Now you maybe thinking, "Oh, so dictionaries and hashtables are basically the same thing, right?" That's not the case. The difference between the 2 is that a dictionary is generic type whereas a hashtable is the opposite. The Hashtable is a weakly typed data structure, so you can add keys and values of any Object Type to the Hashtable. The Dictionary class is a strongly types < T Key, T Value > and you must specify the data types for both the key and value. Dictionaries won't be covered extensively in this activity, but this an interesting piece of information to take note of.
```
Comment (sarahg500) [inline Module4.1_Intro_to_Data_Structures_and_Algos/activities/Act2_HashTables/8.md]: Language is a little too informal here with the first sentence. I would just take it out

File: Module4.1_Intro_to_Data_Structures_and_Algos/activities/Act2_HashTables/7.md [hunk 2]
```diff
@@ -1,19 +1,50 @@
-<!--title={Hash Table as a Dictionary in Python}-->
+### Linear Probing
```
Comment (sarahg500) [inline Module4.1_Intro_to_Data_Structures_and_Algos/activities/Act2_HashTables/7.md]: Add an intro/transitional sentence to the whole card about what they're learning in this card

File: Module4.1_Intro_to_Data_Structures_and_Algos/activities/Act2_HashTables/8.md (hunk 1 above)
Comment (JeffTheAggie) [reply Module4.1_Intro_to_Data_Structures_and_Algos/activities/Act2_HashTables/8.md]: I resolved this issue.

File: Module4.1_Intro_to_Data_Structures_and_Algos/activities/Act2_HashTables/7.md (hunk 2 above)
Comment (JeffTheAggie) [reply Module4.1_Intro_to_Data_Structures_and_Algos/activities/Act2_HashTables/7.md]: I resolved this issue.

File: Module4.1_Intro_to_Data_Structures_and_Algos/activities/Act2_HashTables/8.md (hunk 1 above)
Comment (sarahg500) [reply Module4.1_Intro_to_Data_Structures_and_Algos/activities/Act2_HashTables/8.md]: I was referring to the paragraph above my comment, not the first paragraph. I fixed it myself but please be aware of what my comments refer to and ask questions if you're confused.

**Pull request merged**